wallet/store.go

```diff
@@ -161,42 +165,38 @@ func (k *WalletKey) Open(file *os.File) (w *Wallet, err error) {
 	if err = json.Unmarshal(jsonWallet, ew); err != nil {
 		return nil, err
 	}
+	// this returns an EOF error for some reason
+	//if err = json.NewDecoder(file).Decode(ew); err != nil {
+	//	return
+	//}
```

Comment:
Has the file been read before passing it to `json.NewDecoder` method (e.g. by the `os.ReadFile` a few lines further up)?

In `TestStoreAndRetrieveWalletToFromFile` adding a `file.Seek(0, io.SeekStart)` before calling `wKey.Open` solves the issue.

NIT: this method doesn't need a file as parameter; `io.Reader` would be enough. In the test you could then pass in a `bytes.Buffer` and don't need to actually write anything to a real file. The same can be applied to `Export` by replacing the file with an `io.Writer`.